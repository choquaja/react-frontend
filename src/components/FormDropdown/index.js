import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { groupedElement, formElementStyles } from '../../services/theme/helpers';

const Dropdown = styled.select`
  ${formElementStyles}
  ${props => props.grouped && groupedElement({ borderWidth: '1px', radiusWidth: '2px' })}
`;

const FormDropdown = ({ options, placeholder, ...props }) => (
  <Dropdown {...props}>
    <option value={false} disabled>{placeholder}</option>
    {options.map(({ value, label }) => (
      <option key={value} value={value}>{label}</option>
    ))}
  </Dropdown>
);

FormDropdown.propTypes = {
  options: PropTypes.array.isRequired,
  placeholder: PropTypes.string,
  block: PropTypes.bool,
  full: PropTypes.bool,
  large: PropTypes.bool,
  small: PropTypes.bool,
  grouped: PropTypes.bool,
};

FormDropdown.defaultProps = {
  placeholder: 'Select an option...',
  block: false,
  full: false,
  large: false,
  small: false,
  grouped: false,
};

export default FormDropdown;
