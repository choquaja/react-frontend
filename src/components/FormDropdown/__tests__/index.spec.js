import React from 'react';
import { shallow } from 'enzyme';
import FormDropdown from '../';


const required = {
  options: [],
};
describe('FormDropdown', () => {
  it('renders without crashing', () => {
    shallow(<FormDropdown {...required} />);
  });

  it('renders with a placeholder', () => {
    const formdrop = shallow(<FormDropdown{...required} placeholder="Test" />);
    expect(formdrop).toMatchSnapshot();
  });

  it('renders with block set to true', () => {
    const formdrop = shallow(<FormDropdown{...required} block />);
    expect(formdrop).toMatchSnapshot();
  });

  it('renders with full set to true', () => {
    const formdrop = shallow(<FormDropdown{...required} full />);
    expect(formdrop).toMatchSnapshot();
  });

  it('renders with large set to true', () => {
    const formdrop = shallow(<FormDropdown{...required} large />);
    expect(formdrop).toMatchSnapshot();
  });

  it('renders with small set to true', () => {
    const formdrop = shallow(<FormDropdown{...required} small />);
    expect(formdrop).toMatchSnapshot();
  });

  it('renders with grouped set to true', () => {
    const formdrop = shallow(<FormDropdown{...required} grouped />);
    expect(formdrop).toMatchSnapshot();
  });
});
