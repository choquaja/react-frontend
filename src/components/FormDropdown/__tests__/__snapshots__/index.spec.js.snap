// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDropdown renders with a placeholder 1`] = `
<FormDropdown__Dropdown
  block={false}
  full={false}
  grouped={false}
  large={false}
  small={false}
>
  <option
    disabled={true}
    value={false}
  >
    Test
  </option>
</FormDropdown__Dropdown>
`;

exports[`FormDropdown renders with block set to true 1`] = `
<FormDropdown__Dropdown
  block={true}
  full={false}
  grouped={false}
  large={false}
  small={false}
>
  <option
    disabled={true}
    value={false}
  >
    Select an option...
  </option>
</FormDropdown__Dropdown>
`;

exports[`FormDropdown renders with full set to true 1`] = `
<FormDropdown__Dropdown
  block={false}
  full={true}
  grouped={false}
  large={false}
  small={false}
>
  <option
    disabled={true}
    value={false}
  >
    Select an option...
  </option>
</FormDropdown__Dropdown>
`;

exports[`FormDropdown renders with grouped set to true 1`] = `
<FormDropdown__Dropdown
  block={false}
  full={false}
  grouped={true}
  large={false}
  small={false}
>
  <option
    disabled={true}
    value={false}
  >
    Select an option...
  </option>
</FormDropdown__Dropdown>
`;

exports[`FormDropdown renders with large set to true 1`] = `
<FormDropdown__Dropdown
  block={false}
  full={false}
  grouped={false}
  large={true}
  small={false}
>
  <option
    disabled={true}
    value={false}
  >
    Select an option...
  </option>
</FormDropdown__Dropdown>
`;

exports[`FormDropdown renders with small set to true 1`] = `
<FormDropdown__Dropdown
  block={false}
  full={false}
  grouped={false}
  large={false}
  small={true}
>
  <option
    disabled={true}
    value={false}
  >
    Select an option...
  </option>
</FormDropdown__Dropdown>
`;
